### DISCUSSION TITLE
ifctester, IDS: assign enumeration values to property

### DISCUSSION THREAD

**Discussion by OliverVienna** [IfcOpenShell/IfcOpenShell#3013] [Q&A] [answered]
Hi everyone, is it possible to assign enumeration values to a property instead of a single value?

I have the code block to create an IDS from https://github.com/IfcOpenShell/IfcOpenShell/tree/v0.7.0/src/ifctester. 
And it works fine for a property with a single value.

But I am not sure how to create an IDS with the following property using ifctester: 
Property Set="Pset_ConcreteElementGeneral", Name="CastingMethod", value=["INSITU", "PRECAST"]

I've tried to pass the enumeration values as a list to the class initializer, as follows: 

import ifcopenshell
from ifctester import ids, reporter

```
# Create new IDS
import ifcopenshell
from ifctester import ids, reporter

my_ids = ids.Ids(title="My IDS")

# add specification
my_spec = ids.Specification(name="My first specification")
my_spec.applicability.append(ids.Entity(name="IFCWALL"))
property = ids.Property(
    name="CastingMethod",
    value=["INSITU", "PRECAST"],  
    propertySet="Pset_ConcreteElementGeneral", 
    measure="IfcLabel"
    uri="https://identifier.buildingsmart.org/uri/.../prop/LoadBearing" 
    instructions="The walls must be load-bearing.")
my_spec.requirements.append(property)
my_ids.specifications.append(my_spec)

# Save this IDS in a file
result = my_ids.to_xml("SampleIDS.xml")
```

My question is: Is it even possible to create a property with enumerated values yet?
And if yes, I would appreciate very much any hints on how to do it. 

With kind regards, 
Oliver

**Comment by Moult** [comment, marked as answer]
You need to use a restriction class. See https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifctester/test/test_facet.py for many examples.

You can create properties with enumerated values. Are you asking how to do it in code or how to do it via the Blender interface?

**Reply by OliverVienna** [reply]
Hello @Moult, 
thank you very much for your suggestion. With the examples on test_facet.py and your tip to use a Restiriction class, 
I was able to write a functioning code that looks like this:

```
import ifcopenshell
import ids
from facet import Entity, Attribute, Classification, Property, PartOf, Material, Restriction
import reporter

# Create new IDS
my_ids = ids.Ids(title="My IDS")

# Add specification
my_spec = ids.Specification(name="My first specification")
my_spec.applicability.append(ids.Entity(name="IFCWALL"))

restriction = Restriction(options={"enumeration": ["A00", "A01"]})
facet = ids.Property(
    name="Reference",
    value=restriction,
    propertySet="Pset_WallCommon", 
    measure="IfcIdentifier",
)

# Add property to specification
my_spec.requirements.append(facet)

# Add specification to IDS
my_ids.specifications.append(my_spec)

# Save IDS to file
result = my_ids.to_xml("SampleIDS.xml")
```

I will keep on experimenting, 
Best wishes, 
Oliver
